File: android/src/main/java/expo/modules/audiosession/ExpoAudioSessionModule.kt
package expo.modules.audiosession

import android.content.Context
import android.media.AudioManager
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoAudioSessionModule : Module() {  
  override fun definition() = ModuleDefinition {    
    Name("ExpoAudioSession")
    
    Function("isOtherAudioPlaying") {
      val audioManager = appContext.reactContext?.getSystemService(Context.AUDIO_SERVICE) as? AudioManager
      return audioManager?.isMusicActive ?: false
    }  
  }
}

File: ios/ExpoAudioSessionModule.swift
import ExpoModulesCore
import AVFoundation

public class ExpoAudioSessionModule: Module {  
  public func definition() -> ModuleDefinition {    
    Name("ExpoAudioSession")    
            
    Function("isOtherAudioPlaying") {      
      return AVAudioSession.sharedInstance().isOtherAudioPlaying
    }
  }
}
